build-tools/create-pkg/create-pkg.targets

```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\scripts\installers\create-installers.targets" />
+  <PropertyGroup>
+    <PkgInstallDir>/</PkgInstallDir>
+    <PkgInstallDir Condition="$([MSBuild]::IsOsPlatform(Linux))">/usr</PkgInstallDir>
+    <PayloadDir>$(OutputPath)/pkgarchive</PayloadDir>
+    <PkgOutputPath>$(OutputPath)/packages</PkgOutputPath>
+  </PropertyGroup>
+  <Target Name="_CopyFilesToPayloadDir"
+      DependsOnTargets="GetXAVersion" >
+    <PropertyGroup>
+      <XAFrameworkDir>$(PayloadDir)/Library/Frameworks/Xamarin.Android.framework/Versions/$(XAVersion)</XAFrameworkDir>
+      <MonoFrameworkExternalDir>$(PayloadDir)/Library/Frameworks/Mono.framework/External</MonoFrameworkExternalDir>
+      <XBuildDir>$(XAFrameworkDir)/lib/xamarin.android/xbuild/Xamarin/Android</XBuildDir>
+      <XBuildFrameworkDir>$(XAFrameworkDir)/lib/xamarin.android/xbuild-frameworks/MonoAndroid</XBuildFrameworkDir>
+    </PropertyGroup>
+    <RemoveDir Directories="$(PayloadDir)" />
+    <Copy
+        SourceFiles="@(FrameworkFiles)"
+        DestinationFolder="$(XBuildFrameworkDir)/%(RecursiveDir)"
+    />
+    <Copy
+        SourceFiles="@(FirstFrameworkFiles)"
+        DestinationFolder="$(XBuildFrameworkDir)/$(FirstFrameworkVersion)/%(RecursiveDir)"
+    />
+    <Copy
+        SourceFiles="@(BCLFrameworkFiles)"
+        DestinationFolder="$(XBuildFrameworkDir)/$(BclFrameworkVersion)/%(RecursiveDir)"
+    />
+    <Copy
+        SourceFiles="@(MSBuildFiles);@(MSBuildFilesUnix)"
+        DestinationFolder="$(XBuildDir)/%(RecursiveDir)"
+    />
+    <Copy
+        SourceFiles="@(MSBuildFilesDarwin)"
+        DestinationFolder="$(XBuildDir)/Darwin"
+    />
+    <Copy
+        SourceFiles="@(MSBuildLibFiles)"
+        DestinationFolder="$(XBuildDir)/lib/%(RecursiveDir)"
+    />
+    <Copy
+        SourceFiles="@(MSBuildLibFilesDarwinHost)"
+        DestinationFolder="$(XBuildDir)/lib/host-Darwin"
```

Comment:
Where do these host files need to end up on Unix? Do we need all relevant `host-libs` in this folder, even those for Windows?

We also install (at least a portion of) these files to `/Library/Frameworks/Xamarin.Android.framework/Versions/$(Version)/lib`, do we need to duplicate these here?

Reply:
See e.g. https://github.com/xamarin/xamarin-android/tree/master/build-tools/debian-metadata, which appears to just include *everything* in the `bin/%CONFIG%/lib/xamarin.android` directory, with no exclusions.

We might not "need" them, but we "include" them.

Note furthermore, though, that `host-Darwin` *will not exist* on Linux builds. It'll be `host-Linux`.

This step should probably use `host-$(HostOS)`, which will (mostly) Do The Right Thing™ based on host OS.

As such, this should probably be:


```xml
    <Copy
        Condition=" '$(HostOS)' != 'Windows' "
        SourceFiles="@(MSBuildLibFilesUnixHost)"
        DestinationFolder="$(XBuildDir)/lib/host-$(HostOS)"
    />
```

Reply:
> We also install (at least a portion of) these`host-*` libraries to `/Library/Frameworks/Xamarin.Android.framework/Versions/$(Version)/lib`, is that necessary?

Good question.  I don't remember anymore. :-(

Reply:
@jonpryor on this note it looks like these `/Library/Frameworks/Xamarin.Android.framework/Versions/$(Version)/lib` `host` libraries are created or "linked" via an `install_name_tool` call. I suppose this can also be recreated for purposes of backwards compatibility if you feel that it is necessary.

Reply:
The assumptions were made about a 'Darwin' folder existing for all unix invocations should now be fixed by using the $(HostOS) property value where applicable. A 'host' symlink has been added, and all 'host' library files are now copied to '$(XAFrameworkDir)/lib' to ensure compatibility with previous .pkg creation logic.

The `install_name_tool` invocation has been ignored intentionally for now. All `host-*/*.d.*` files are excluded as well, and only the `host-*` files which pertain to the relevant installation environment are included (e.g. `host-mxe-Win*` files are now excluded on Unix).